src/components/tabs/js/tabsController.js

```diff
@@ -97,6 +99,92 @@ function MdTabsController ($scope, $element, $window, $mdConstant, $mdTabInkRipp
     delete $attrs.$mdTabsTemplate;
   }
 
+  var DOMtabs = [];
+
+  function handleScroll () {
+    var nbTabs = ctrl.tabs.length;
+
+    var watcher = $scope.$watch(function () {
+      return $element.find('md-tab-content').length;
+    }, function (nv) {
+      if (nv === nbTabs) {
+        watcher();
+        refreshDOMtabs();
+      }
+    });
+  }
+
+  function refreshDOMtabs () {
+    DOMtabs = $element.find('md-tab-content');
+    if (ctrl.shrinkToolbar) {
+      bindScrollListener(ctrl.selectedIndex);
+    }
+  }
+
+  function bindScrollListener (newIndex, oldIndex) {
+    if (!DOMtabs.length) return;
+    if (oldIndex !== undefined && ctrl.tabs[oldIndex]) {
+      angular.element(DOMtabs[oldIndex]).off('scroll');
+      $timeout(function () {
+        scrollTop(oldIndex);
+      }, 1000);
+    }
+
+    prevScrollTop = 0;
+    angular.element(DOMtabs[newIndex || 0])
+      .off('scroll')
+      .on('scroll', shrinkToolbar);
+  }
+
+  var prevScrollTop = 0;
+  var shrinkSpeedFactor = .5;
+  var contentScroll = 0;
+  var toolbar;
+  var toolbarHeight = 0;
+  var y = 0;
+  var ZCLASS = 'md-whiteframe-z1';
+
+  function shrinkToolbar (e) {
+    if (!toolbar) {
+      elements.toolbar = angular.element($element.parent()[0].querySelector('md-toolbar'));
+    }
+
+    toolbarHeight = elements.toolbar.prop('offsetHeight');
+
+    var isScrollingDown = scrollTop >= prevScrollTop;
+    var scrollTop = e.target.scrollTop;
+    var margin = toolbarHeight * shrinkSpeedFactor;
+
+    y = Math.min(
+      toolbarHeight / shrinkSpeedFactor,
+      Math.max(0, y + scrollTop - prevScrollTop)
+    );
+
+    if (!isScrollingDown || (isScrollingDown && contentScroll > 0)) {
+
+      y = scrollTop === 0 ? 0 : y;
+
+      contentScroll = (toolbarHeight - y) * shrinkSpeedFactor;
+
+      elements.toolbar.css('transform', 'translateY(' + (-y * shrinkSpeedFactor) + 'px)');
+      $element.css('transform', 'translateY(' + contentScroll + 'px)');
+
+      $element.css({
+        "margin-top": - margin + 'px',
+        "margin-bottom": ((toolbarHeight - y) * shrinkSpeedFactor) + 'px'
+      });
+
+      if (contentScroll >= toolbarHeight - y ) {
+        angular.element(elements.wrapper).addClass(ZCLASS);
+      } else {
+        angular.element(elements.wrapper).removeClass(ZCLASS);
+      }
```

Comment:
This whole expression is the same as `angular.element(elements.wrapper).toggleClass(ZCLASS, contentScroll >= toolbarHeight - y);`